lib/api/stream.ex

```diff
@@ -0,0 +1,108 @@
+defmodule Api.Stream do
+  @moduledoc """
+  A GenStage for connecting to the API's Server-Sent Event Stream
+  capability. Receives events from the API and parses their data.
+  Subscribers receive events as `%Api.Stream.Event{}` structs, which
+  include the event name and the data as a `%JsonApi{}` struct.
+
+  Required options:
+  `:path` (e.g. "/vehicles")
+  `:name` -- name of module
+  `:subscribe_to` -- pid or name of a ServerSentEventStage
+  for the Api.Stream to subscribe to. This should be
+  started as part of a supervision tree.
+
+  Other options are made available for tests, and can include:
+  - :name (name of the GenStage process)
+  - :base_url
+  - :api_key
+  """
+
+  use GenStage
+  alias ServerSentEventStage, as: SSES
+
+  defmodule Event do
+    @moduledoc """
+    Struct representing a parsed Api server-sent event.
+    """
+    defstruct data: nil, event: :unknown
+    @type event :: :reset | :add | :update | :remove
+    @type t :: %__MODULE__{
+            event: event | :unknown,
+            data: nil | JsonApi.t() | {:error, any}
+          }
```

Comment:
Nothing makes the `data: nil` case or the `event: :unknown` case. They could be removed, and the fields marked by `@enforce_keys`.